Doc/library/ctypes.rst

```diff
@@ -1799,10 +1799,31 @@ different ways, depending on the type and number of the parameters in the call:
    integer. *name* is name of the COM method. *iid* is an optional pointer to
    the interface identifier which is used in extended error reporting.
 
+   If *iid* is not specified, a :exc:`WindowsError` is raised if the COM method
+   call fails. If *iid* is specified, a :exc:`.COMError` is raised instead.
+
    COM methods use a special calling convention: They require a pointer to
    the COM interface as first argument, in addition to those parameters that
    are specified in the :attr:`!argtypes` tuple.
 
+
+.. exception:: COMError(hresult, text, details)
+
+   Windows only: This non-public exception is raised when a COM method call
```

Comment:
You can use `.. availability:: Windows` (but put the directive after the class doc)

Reply:
I followed the pattern in `ctypes.rst` because it frequently used sentences like `Windows only: ...`.

Since this markup wasn't mentioned in [the documentation guide](https://devguide.python.org/documentation/markup/#), I overlooked it.
Is this a more modern approach?

Reply:
It's widely used in the https://docs.python.org/3/library/socket.html module actually.

Reply:
Referring to the [`socket.ioctl`](https://docs.python.org/3/library/socket.html#socket.socket.ioctl) documentation, it seems that using `:platform: Windows` might be more appropriate than `.. availability:: Windows`. What do you think?

Alternatively, could it be that `:platform:` is intended to be used exclusively in module sections, as described in the [documentation guide's "Module-specific markup"](https://devguide.python.org/documentation/markup/#module-specific-markup)?